boards/posix/native_posix/Kconfig.defconfig

```diff
@@ -103,3 +103,14 @@ config USB_NATIVE_POSIX
 	default y
 
 endif # USB
+
+# i386 assembly code used in MBEDTLS does not compile with size optimization
+# if address sanitizer is enabled, as such switch default optimization level
+# to speed
+if ASAN && !64BIT && MBEDTLS
+
+choice OPTIMIZATION_LEVEL
+	default SPEED_OPTIMIZATIONS
```

Comment:
It would then be the same issue with the nrf52_bsim (even if we typically do not test mbedtls there)
Maybe it would be easier to just set the default optimization level to no_opt with ASAN just like with COVERAGE?

Reply:
I didn't want to nitpick, but if we are going to change this then it would be cleaner to just modify the build environment for the mbedtls library, and not for the entire build.

Reply:
An option would be to apply the following patch to `crypto/mbedtls/CMakeLists.txt`

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index d382aea..ccf9a61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,6 +94,10 @@ if(CONFIG_MBEDTLS_BUILTIN)
   zephyr_library_sources(library/x509write_csr.c)
   zephyr_library_sources(library/xtea.c)

+if(CONFIG_ARCH_POSIX AND CONFIG_ASAN AND NOT CONFIG_64BIT)
+  set_property(SOURCE library/bignum.c APPEND PROPERTY COMPILE_OPTIONS "-O0")
+endif ()
+
   zephyr_library_link_libraries(mbedTLS)
 else()
   assert(CONFIG_MBEDTLS_LIBRARY "MBEDTLS was enabled, but neither BUILTIN or LIBRARY was selected.")

```

Reply:
Change from -O0 to ${OPTIMIZE_FOR_SPEED_FLAG} or ${OPTIMIZE_FOR_NO_OPTIMIZATIONS_FLAG} and this looks good to me.

Reply:
changes moved to mbedtls module